wgengine/wgcfg/config.go

```diff
@@ -35,6 +36,20 @@ type Config struct {
 	}
 }
 
+func (c *Config) Equal(o *Config) bool {
+	if c == nil || o == nil {
+		return c == o
+	}
+	return c.Name == o.Name &&
+		c.NodeID == o.NodeID &&
+		c.PrivateKey.Equal(o.PrivateKey) &&
+		c.MTU == o.MTU &&
+		c.NetworkLogging == o.NetworkLogging &&
+		slices.Equal(c.Addresses, o.Addresses) &&
+		slices.Equal(c.DNS, o.DNS) &&
+		slices.EqualFunc(c.Peers, o.Peers, func(a, b Peer) bool { return a.Equal(b) })
```

Comment:
I think we don't need to curry manually here, do we?
```suggestion
		slices.EqualFunc(c.Peers, o.Peers, Peer.Equal)
```

Reply:
Embarrassing! :) Fixed.

Reply:
Static tools should check for this :)

I just filed dominikh/go-tools#1681